gui/directoryservice/forms.py

```diff
@@ -712,48 +713,53 @@ def save(self):
         return obj
 
 
-class NISForm(ModelForm):
+class NISForm(MiddlewareModelForm, ModelForm):
+
+    middleware_attr_prefix = 'nis_'
+    middleware_attr_schema = 'nis'
+    middleware_plugin = 'nis'
+    is_singletone = True
+
     class Meta:
         fields = '__all__'
         model = models.NIS
 
     def __init__(self, *args, **kwargs):
         super(NISForm, self).__init__(*args, **kwargs)
+        self.original_instance = dict(self.instance.__dict__)
         self.fields["nis_enable"].widget.attrs["onChange"] = (
             "nis_mutex_toggle();"
         )
 
     def save(self):
         enable = self.cleaned_data.get("nis_enable")
-
-        # XXX: We need to have a method to test server connection.
-        try:
-            started = notifier().started("nis")
-        except Exception:
-            raise MiddlewareError(_("Failed to check NIS status."))
-        finally:
-            super(NISForm, self).save()
-
+        with client as c:
+            state = c.call('nis.get_state')
```

Comment:
I am thinking all this logic about start and start/restart service should be bundled into nis.update method?

Otherwise this will be left to the new UI team to figure out and more logic in the client side.

Reply:
Yes. I agree.